tests/RobotFramework/config_management/child_conf_mgmt_plugin.robot

```diff
@@ -0,0 +1,333 @@
+    Should Be Equal    ${rc}    ${0}
+    Close Connection
+Request snapshot from child device
+    ${json_snap}=    Set Variable    {"deviceId":"${child_id}","description":"Retrieve config1 configuration snapshot from device ${CHILD}","c8y_UploadConfigFile":{"type":"config1"}}
+    Connect    ${PARENT_IP}
+    @{messages}=    Subscribe    tedge/${CHILD}/commands/req/config_snapshot    qos=1    timeout=0   limit=0
+    ${auth}=    Create List    ${user}    ${pass}
+    Create Session    API_Testing    https://${url_tedge}    auth=${auth}
+    ${Get_Response}=    POST On Session    API_Testing    /devicecontrol/operations    ${json_snap}    #expected_status=200
+    @{listen}=    Listen    tedge/${CHILD}/commands/req/config_snapshot    timeout=20    limit=1
+    Should Be Equal    @{listen}    {"url":"http://${PARENT_IP}:${HTTP_PORT}/tedge/file-transfer/${CHILD}/config_snapshot/config1","path":"/home/pi/config1","type":"config1"}
+    [Teardown]    Disconnect
+    #CHECK OPERATION 
+    ${auth}=    Create List    ${user}    ${pass}
+    Create Session    API_Testing    https://${url_tedge}    auth=${auth}
+    ${Get_Response}=    GET On Session    API_Testing    /devicecontrol/operations
+    ${json_response}=    Set Variable    ${Get_Response.json()}
+    @{pd_name}=    Get Value From Json    ${json_response}    $..status
+    ${first}    Get From List    ${pd_name}    0
+    ${second}    Get From List    ${pd_name}    1
+    ${third}    Get From List    ${pd_name}    2
+    ${fourth}    Get From List    ${pd_name}    3
+    Should Be Equal    ${first}    PENDING
+Child device response on snapshot request    
+    Child Connection
+    ${rc}=    Execute Command    mosquitto_pub -h ${PARENT_IP} -t "tedge/${CHILD}${topic_snap} -m ${payl_exec}    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0}
+    ${rc}=    Execute Command    curl -X PUT --data-binary @/home/pi/config1 http://${PARENT_IP}:${HTTP_PORT}/tedge/file-transfer/${CHILD}/config_snapshot/config1    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0}
+    Sleep    5s
+    ${rc}=    Execute Command    mosquitto_pub -h ${PARENT_IP} -t "tedge/${CHILD}${topic_snap} -m ${payl_succ}    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0}
+    Sleep    2s
+    Close Connection
+    #CHECK OPERATION 
+    ${auth}=    Create List    ${user}    ${pass}
+    Create Session    API_Testing    https://${url_tedge}    auth=${auth}
+    ${Get_Response}=    GET On Session    API_Testing    /devicecontrol/operations
+    ${json_response}=    Set Variable    ${Get_Response.json()}
+    @{pd_name}=    Get Value From Json    ${json_response}    $..status
+    ${first}    Get From List    ${pd_name}    0
+    ${second}    Get From List    ${pd_name}    1
+    ${third}    Get From List    ${pd_name}    2
+    ${fourth}    Get From List    ${pd_name}    3
+    Should Be Equal    ${first}    SUCCESSFUL
```

Comment:
This block seems to be repeating multiple times. Can this be extracted into a common routine?

Reply:
Difference is the expected result in the Should be equal, it can be of course in separate keyword but than I need to make the variables on suite level, and this will complicate the definitions so I decided to have it like this, and because they are in the keyword session they are reusable. But if there is some relevant concern and not "cosmetic reason" I can reconsider it